### PR TITLE
Handle manual dependabot workflow trigger 

### PR DESCRIPTION
**Pull request by figi44**
Includes changes to git commands to avoid force pushing and handling for manual workflow trigger

### CODE REVIEW

File: .github/workflows/dependabot-build.yml [hunk 1]
```diff
@@ -25,6 +25,13 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
+      - name: Git config
+        run: |
+          git config --global user.email "github-actions@example.com"
+          git config --global user.name "github-actions[bot]"
+          git fetch origin ${{ github.head_ref }}
+          git checkout ${{ github.head_ref  }}
```
Comment (dtip) [inline .github/workflows/dependabot-build.yml]: @figi44 why do we need to fetch & checkout here when we already using `actions/checkout` above?

Comment (figi44) [reply .github/workflows/dependabot-build.yml]: good point, the same could be achieved with a `ref` input to `actions/checkout`

Comment (dtip) [reply .github/workflows/dependabot-build.yml]: Does `actions/checkout` not automatically checkout the correct ref?

Comment (dtip) [reply .github/workflows/dependabot-build.yml]: Had a chat about it - `actions/checkout` will checkout the internal GitHub merge commit for the PR, no the branch we need, so passing the `ref` in makes sense.

**Pull request merged**